src/main/java/org/apache/commons/cli/converters/Converter.java

```diff
@@ -0,0 +1,74 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package org.apache.commons.cli.converters;
+
+import java.io.File;
+import java.net.URL;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * The definition of the functional interface to call when doing a conversion.
+ * Like {@code Function<String,T>} but can throw an Exception.
+ *
+ * @param <T> The return type for the function.
+ * @since 1.7
+ */
+@FunctionalInterface
+public interface Converter<T> {
```

Comment:
Probably should extend `Function<String, T>`.

Reply:
Right, we don't want to depend on Commons Lang's `FailableException` here.

Next: Throwing `Exception` is an anti-pattern and should be avoided at all costs IMO. Either:

- reconsider and throw runtime exception OR 
- throw a more specific checked exception OR
- parameterize the exception type
-